File: src/types/workspace.ts
export interface Workspace {
  id: string;
  name: string;
  icon: string;
  working_directory: string;
  created_at: string;
  updated_at: string;
}

export interface CreateWorkspaceRequest {
  name: string;
  icon?: string;
  working_directory?: string;
  agent_ids?: string[];
}

export interface UpdateWorkspaceRequest {
  name?: string;
  icon?: string;
  working_directory?: string;
}

File: src-tauri/src/models/workspace.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Workspace {
    pub id: String,
    pub name: String,
    pub icon: String,
    pub working_directory: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateWorkspaceRequest {
    pub name: String,
    #[serde(default = "default_icon")]
    pub icon: String,
    #[serde(default)]
    pub working_directory: String,
    #[serde(default)]
    pub agent_ids: Vec<String>,
}

fn default_icon() -> String {
    "folder".into()
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateWorkspaceRequest {
    pub name: Option<String>,
    pub icon: Option<String>,
    pub working_directory: Option<String>,
}
